streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -507,4 +510,51 @@ public void shouldLockGlobalStateDirectoryWhenDirectoryCreationDisabled() throws
         initializeStateDirectory(false);
         assertTrue(directory.lockGlobalState());
     }
+
+    @Test
+    public void shouldNotFailWhenCreatingTaskDirectoryInParallel() throws Exception {
```

Comment:
Without the fix, this test fails like 100% of the time for me. (Kinda surprising -- seems that the disk IO operations give us nice thread context switches so we hit the issue reliably.)

Reply:
Huh, neat!

Reply:
In retrospect, this makes sense. The fs operations probably take an order of magnitude more time than the difference in time between starting the two threads. Thanks for adding it!